src/astro/files/base.py

```diff
@@ -34,13 +34,15 @@ def __init__(
             path=path, filetype=filetype, normalize_config=normalize_config
         )
 
+    # To Do: path property can resolve into lists of path(list[str]) or single path(str).
+    # Right now code is assuming value of string which is wrong need to make changes all across code.
     @property
-    def path(self) -> str:
+    def path(self) -> Any:
```

Comment:
It may be worth being more specific about this, using `Union` of list of strings and strings

Reply:
+1

Reply:
@tatiana  We need to use typing.cast() for this. Updated the code.